aospy/utils/times.py

```diff
@@ -198,13 +198,14 @@ def numpy_datetime_workaround_encode_cf(ds):
     time = ds[internal_names.TIME_STR]
     units = time.attrs['units']
     units_yr = units.split(' since ')[1].split('-')[0]
+    ref_yr = int(units_yr)
     min_yr_decoded = xr.decode_cf(time.to_dataset(name='dummy'))
     min_date = min_yr_decoded[internal_names.TIME_STR].values[0]
     if isinstance(min_date, np.datetime64):
         return ds, pd.Timestamp(min_date).year
     else:
         min_yr = min_date.year
```

Comment:
After defining `min_yr`, let's define `offset = int(units_yr) - min_yr + 1` rather than `ref_yr` and include a comment above it describing it, e.g. `# Offset such that decoded dates start in the earliest valid whole year`.  Then the line below becomes `new_units_yr = pd.Timestamp.min.year + offset`.